### PR TITLE
Adds Etcdv2

### PR DESCRIPTION
**Pull request by JoshVanL**
See https://github.com/dapr/components-contrib/pull/2932

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 1]
```diff
@@ -38,6 +40,16 @@ spec:
 The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
 {{% /alert %}}
 
+
+### Versioning
+
+Dapr has 2 versions of the Etcd state store component: `v1` and `v2`.
+
+`v1` and `v2` have the same metadata fields, however `v1` will cause data inconsistencies in apps when using [Actor TTLs]({{< ref "actors_api.md#ttl" >}}) from Dapr `v1.12`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
While `v1` and `v2` have the same metadata fields, `v1` causes data inconsistencies in apps when using [Actor TTLs]({{< ref "actors_api.md#ttl" >}}) from Dapr v1.12.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 2]
```diff
@@ -38,6 +40,16 @@ spec:
...
+Users should always use `v2` over `v1` as `v1` is deprecated.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
```

Moved this up ^^

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 3]
```diff
@@ -38,6 +40,16 @@ spec:
 The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
 {{% /alert %}}
 
+
+### Versioning
+
+Dapr has 2 versions of the Etcd state store component: `v1` and `v2`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
Dapr has 2 versions of the Etcd state store component: `v1` and `v2`. It is recommended to use `v2`, as `v1` is deprecated.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 4]
```diff
@@ -38,6 +40,16 @@ spec:
...
+
+`v1` and `v2` have the same metadata fields, however `v1` will cause data inconsistencies in apps when using [Actor TTLs]({{< ref "actors_api.md#ttl" >}}) from Dapr `v1.12`.
+Users should always use `v2` over `v1` as `v1` is deprecated.
+`v1` and `v2` are incompatible and there is no data migration path for `v1` to `v2` on an existing active Etcd cluster and `keyPrefixPath`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
`v1` and `v2` are incompatible with no data migration path for `v1` to `v2` on an existing active Etcd cluster and `keyPrefixPath`.
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 5]
```diff
@@ -38,6 +40,16 @@ spec:
...
+If you are using `v1`, you should continue to use `v1` until you create a new Etcd cluster or use a different `keyPrefixPath`, then use `v2`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
If you are using `v1`, you should continue to use `v1` until you create a new Etcd cluster or use a different `keyPrefixPath`.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review

**Comment by JoshVanL** [comment]
Thanks @hhunter-ms!

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**